src/disp/syscall.c

```diff
@@ -1270,6 +1270,28 @@ static MVMDispSysCall async_unix_listen = {
     .expected_concrete = { 1, 1, 1, 1, 0 },
 };
 
+/* handle-open-mode */
+static void handle_open_mode_impl(MVMThreadContext *tc, MVMArgs arg_info) {
+    MVMOSHandle *oshandle  = (MVMOSHandle *)get_obj_arg(arg_info, 0);
+
+    MVMIOFileData *data = (MVMIOFileData *)oshandle->body.data;
+    MVMint64 open_mode  = (MVMint64) recover_open_mode_simple(data->open_mode_simple);
+
+    if (open_mode < 0)
+        MVM_exception_throw_adhoc(tc, "Incomprehensible open mode");
```

Comment:
Could this error message include some more context? E.g., something like “, %d is < 0” at the end.